src/Thermostat.js

```diff
@@ -106,12 +107,22 @@ export default class Thermostat extends EventEmitter {
   }
   setTargetHeatingCoolingState(state) {
     return this.connect().then(() => {
-      switch (state) {
-        case TargetHeatingCoolingState.OFF: return this.device.turnOff()
-        case TargetHeatingCoolingState.HEAT: return this.device.turnOn()
-        case TargetHeatingCoolingState.AUTO: return this.device.automaticMode()
-        case TargetHeatingCoolingState.COOL: return this.device.manualMode()
-        default: throw new Error('Unsupported mode')
+      if (this.manualOnly) {
+        switch (state) {
```

Comment:
Please refactor this `switch` statement, so that the check, if `manualOnly` is active, is inside the `case` statement. With that, we don't have to duplicate the whole `switch`